src/hpp/rostools.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import print_function
+
+try: # python 2
+    _basestr = basestring
+    encoding = { 'encoding': 'utf-8' }
+except NameError: # python 3
+    _basestr = str
+    unicode = str
+    encoding = {}
+
+def process_xacro(*args):
+    import xacro
+
+    opts, input_file_name = xacro.process_args(args)
+    try:
+        # open and process file
+        doc = xacro.process_file(retrieve_resource(input_file_name), **vars(opts))
+
+    # error handling
+    except xml.parsers.expat.ExpatError as e:
+        xacro.error("XML parsing error: %s" % unicode(e), alt_text=None)
+        if xacro.verbosity > 0:
+            xacro.print_location(filestack, e)
+            print(file=sys.stderr) # add empty separator line before error
+            print("Check that:", file=sys.stderr)
+            print(" - Your XML is well-formed", file=sys.stderr)
+            print(" - You have the xacro xmlns declaration:",
+                  "xmlns:xacro=\"http://www.ros.org/wiki/xacro\"", file=sys.stderr)
+        sys.exit(2)  # indicate failure, but don't print stack trace on XML errors
+
+    except Exception as e:
+        msg = unicode(e)
+        if not msg: msg = repr(e)
+        xacro.error(msg)
+        if xacro.verbosity > 0:
+            xacro.print_location(filestack, e)
+        if xacro.verbosity > 1:
+            print(file=sys.stderr)  # add empty separator line before error
+            raise  # create stack trace
+        else:
+            sys.exit(2)  # gracefully exit with error condition
+
+    # write output
+    return doc.toprettyxml(indent='  ', **encoding)
+
+def retrieve_resource(path):
+    if path.startswith("package://"):
+        relpath = path[len("package://"):]
+        import os
+        rospaths = os.environ["ROS_PACKAGE_PATH"].split(':')
```

Comment:
I think this function should be able to work even if `ROS_PACKAGE_PATH` is not defined in the current environment. But we can still merge this PR like this for now.

Reply:
If you have `package://` but no `ROS_PACKAGE_PATH`, what behavior do you expect ?